api/v1alpha1/regexp_test.go

```diff
@@ -0,0 +1,294 @@
+package v1alpha1_test
+
+import (
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"io/fs"
+	"regexp"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/akuity/kargo/api/v1alpha1/testhelper"
+)
+
+// ValidateResourceExpression checks the regular expression identified by marker
+// against the provided test cases.
+func ValidateResourceExpression(t *testing.T, marker string, testCases map[string]bool) {
+	expression, err := findExpression(marker)
+	require.NoError(t, err)
+	testhelper.ValidateRegularExpression(t, expression, testCases)
+}
+
+func Test_Pattern_Branch(t *testing.T) {
```

Comment:
Nit: I'm usually reluctant to use snake case in test function names. I'll do it, but only when I feel _forced_ to because, for instance, multiple types in the same package have functions with the same name and `Test_Foo_Hello()` and `Test_Bar_Hello()` end up being _much_ more clear than `TestFooHello()` and `TestBarHello()`, which would easily fool anyone into thinking we're testing functions named `FooHello()` and `BarHello()`.

This is a preference and not even close to a hill I would die on, but I'd like to hear from @hiddeco about how strongly (or not) he feels this should be `TestBranchPattern()`.

Obviously applies to other functions in this same file.

Reply:
I only use snake cases for method testing (e.g. `TestFoo_Bar` for method `Bar` from `Foo`), or in case the function I am testing is private (e.g. `Test_foo` for function `foo()`).

Reply:
These are easy enough to change because nothing calls them. Fix incoming and then this LGTM.

Reply:
That was just me not quite understanding the use of underscores when I started this little endeavour. Thanks.